cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1325,6 +1327,16 @@ data BuildTxWith build a where
      ViewTx      ::      BuildTxWith ViewTx  a
      BuildTxWith :: a -> BuildTxWith BuildTx a
 
+defaultBuildTxWith :: ModeTx build -> a -> BuildTxWith build a
+defaultBuildTxWith inModeTx a =
+  case inModeTx of
+    InViewTx -> ViewTx
```

Comment:
We aren't interested in building a `ViewTx` so we can totally ignore it and do away with the `ModeTx a` type.